✨Adicionar tags

YanYamim opened the pull request:
<h2>O que foi feito até agora</h2>
<h3>Frontend</h3>
<ul>

 <li>Botão de criação de Tags</li>
 <li> Campos com o nome, descrição e data de criação</li>
 <li>3 Tags de exemplificação</li>
 <li>Botão de edição, onde é permitido alterar nome e descrição da tag</li>

</ul>
<p>Link do .vue do front http://localhost:5173/</p>

<h3>Backend</h3>
<ul>

 <li>Controller das tags com GET, POST e PUT (Todas testadas no Insomnia)</li>
 <li> Classe Tag com os atributos: Id, nome, descrição e um Boolean ativo</li>
 <li>Classes para cadastro, atualização e listagem dos dados das tags</li>

</ul>

<h2>O que pode melhorar</h2>
<ul>

 <li>Testar as implementações para confirmar se está tudo certo</li>
 <li>As tags serem salvas de modo que se sairmos da página e voltarmos as tags cadastradas estarão lá</li>
 <li>As edições das tags serem salvas</li>
 <li>Otimização dos códigos (Se precisar)</li>

</ul>

Yan-Yamim committed:
DRAFT

Yan-Yamim committed:
🚧 Começo do back do cadastro de tags

ok

Yan-Yamim committed:
⚡️ Correção Java, html e css e devcontainer

ok

Yan-Yamim committed:
🚧 Implementação do frontend com Vue

ok

Yan-Yamim committed:
🐛 Corriji problemas no front e back

ok

Yan-Yamim committed:
✨ Adicionei campo de data de criação (dia de hoje)

ok

Yan-Yamim committed:
🚸 Implementei a checkbox de ativo nas tags

ok

marilia-borgo commented on frontend/src/App.vue:
```diff
@@ -1,85 +1,229 @@
-<script setup lang="ts">
-import { RouterLink, RouterView } from 'vue-router'
-import HelloWorld from './components/HelloWorld.vue'
+<template>
```
Eu e o @YanYamim Conversamos presencialmente na fatec hoje, e eu expliquei sobre o conceito de componentização (components e views).
Ele vai editar essa parte e adicionar o router também

marilia-borgo commented on .idea/FATEC-API-3-SEMESTRE.iml:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Tem que apagar tudo o que está nessa página .idea
Não sei porque o gitgnore não barrou, mas de qualquer jeito tem de sair.

marilia-borgo commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,23 @@
+{
```
tudo dentro dessa pasta também tem que morrer. 

Vou adicionar o .devconteiner no gitignore

c137santos replied:
Você precisa atender esse pedido da marilia

marilia-borgo commented on .idea/jarRepositories.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
O que essa pasta faz?
Ele devia estar aqui? Acho que foi criado por alguma IDE, testa apagar esse arquivo e rodar o projeto. Caso funcione tire a pasta inspectionProfiles do pr também.

marilia-borgo commented on frontend/package.json:
```diff
@@ -12,6 +12,7 @@
     "format": "prettier --write src/"
   },
   "dependencies": {
+    "frontend": "file:",
```
Pra que serve isso? 
Dúvida sincera

Yan-Yamim committed:
🦺 Adicionei componentes a validação do Vue

ok

Yan-Yamim committed:
🔥 Remoção de arquivos desnecessários e rodou

ok

Yan-Yamim committed:
🐛 Corriji o bug da logo

ok

Yan-Yamim committed:
🐛 Corriji a posição do botão de cadastrar tags

ok

c137santos commented on .devcontainer/docker-compose.yml:
atualizar com o main-primeira-psrint

c137santos commented on backend/pom.xml:
Isso tudo aqui faz parte da minha task, então ao resolver o conflito, preferi as alterações que existem na branch main-primeira-sprint

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-data-jpa</artifactId>
		</dependency>

		<dependency>
			<groupId>org.postgresql</groupId>
			<artifactId>postgresql</artifactId>
			<scope>runtime</scope>
		</dependency>

		<dependency>
			<groupId>com.h2database</groupId>
			<artifactId>h2</artifactId>
			<scope>runtime</scope>
		</dependency>

c137santos commented on backend/src/main/java/com/group/backend/domain/Tags/Tag.java:
Isso aqui faz parte do meu card, retirar isso aqui e preferir as alterações do main-primeira-sprint

Yan-Yamim committed:
✅ Arrumei arquivos que conflitavam

ok

Yan-Yamim committed:
➕ Arrumei os conflitos no Maven

ok

Yan-Yamim committed:
🐛 Arrumando conflitos

ok

Yan-Yamim committed:
🐛 Resolvendo conflitos

ok

c137santos commented on Dockerfile.dev:
Esse aqui vc não pode deletar kkk volta ele. O que era para deletar era aquela pasta com os arquivos que definiam as extensões que vc tinha construido. Volta o [Dockerfile.dev](https://github.com/c137santos/FATEC-API-3-SEMESTRE/blob/main/Dockerfile.dev)

c137santos commented on backend/pom.xml:
```diff
@@ -61,7 +99,16 @@
 				<artifactId>versions-maven-plugin</artifactId>
 				<version>2.17.1</version>
 			</plugin>
+			<plugin>
```
Eu não to vendo problema em você mergear isso, pq eu comparei os dois arquivos e estão iguais, Mas na resolução de conflito onde vc escolhe suas modificações, ao invés das modificações que da main, corretamente, corre o risco de você sobrescrever coisas. Então é preciso saber sempre, o que é seu, e o que é da main, e como isso deve conviver.

Yan-Yamim committed:
Merge branch 'main-first-sprint' into 10-adicionar-tags

Yan-Yamim committed:
🚧 Correção do backend e inicio da conexão BD

ok

MarcyLeite commented:
A esse ponto, testado e rodando com o Postgres em docker

MarcyLeite committed:
Merge branch 'main-first-sprint' into 10-adicionar-tags

Yan-Yamim committed:
🎨 Correção de estruturas

ok

Yan-Yamim committed:
🐛 Resolvendo conflito do pom.xml

ok

Yan-Yamim committed:
🎨 Alterações do front para o banco

Alterações da função cadastrar tag para deixá-la mais direta e um crossOrigin (*) para testes com o banco

Yan-Yamim committed:
⚡️ Implementação WebConfig

Implementei um WebConfiguration para conseguir conectar o back com o front

Commit:
🐛 Corrigindo lógica do front com comunicação com back

Havia um bug do botão não está realizando solicitação, e quando fazia, estava com os dados vazios.

Yan-Yamim committed:
✨ Ligação com o banco funcionando
